modules/ingester/instance.go

```diff
@@ -488,12 +548,223 @@ func (i *instance) rediscoverLocalBlocks(ctx context.Context) error {
 			return err
 		}
 
+		//sb := search.OpenBackendSearchBlock(i.local, b.BlockMeta().BlockID, b.BlockMeta().TenantID)
+
 		i.blocksMtx.Lock()
 		i.completeBlocks = append(i.completeBlocks, ib)
+		//i.searchCompleteBlocks[ib] = sb
 		i.blocksMtx.Unlock()
 
 		level.Info(log.Logger).Log("msg", "reloaded local block", "tenantID", i.instanceID, "block", id.String(), "flushed", ib.FlushedTime())
 	}
 
 	return nil
 }
+
+func (i *instance) Search(ctx context.Context, req *tempopb.SearchRequest) ([]*tempopb.TraceSearchMetadata, error) {
+
+	done := make(chan struct{})
+	defer close(done)
+
+	var resultsCh []<-chan *tempopb.TraceSearchMetadata
```

Comment:
Nice addition. Should we add an `errorCh` as well? Each of the individual search functions could fail silently.

Reply:
True.  We could bubble errors up and log them, but not sure where that is most appropriate.   The current instance search method doesn't wait for tasks to complete, and may not be around to log the errors.

Reply:
I'm thinking in the `instance.Search()` method, we could `select` on a `resultsCh` and `errorCh`. [At line 32](https://github.com/grafana/tempo/pull/806/files#diff-8c9aac72ec16ad39eb093753942cde7f893be0b5950578ffe10ef003d2ff5237R32), once all the workers have been fired off, instead of
```
	for result := range sr.Results() {
```
we can have something like:

```
for {
  select {
    case <- sr.Results():
      // handle results
    case <- sr.Errors():
      // handle errors
  }
}
```